File: app-presentation/src/main/java/com/zxjdev/smile/presentation/common/base/fragment/BaseFragment.java
package com.zxjdev.smile.presentation.common.base.fragment;

import com.zxjdev.smile.presentation.common.util.image.GlideImageLoader;
import com.zxjdev.smile.presentation.common.util.image.ImageLoader;

import dagger.android.support.DaggerFragment;

public class BaseFragment extends DaggerFragment {

  private ImageLoader imageLoader;

  protected ImageLoader getImageLoader() {
    if (imageLoader == null) {
      imageLoader = new GlideImageLoader(this);
    }
    return imageLoader;
  }
}

File: app-presentation/src/main/java/com/zxjdev/smile/presentation/common/base/fragment/BaseFragment.kt
package com.zxjdev.smile.presentation.common.base.fragment

import com.zxjdev.smile.presentation.common.util.image.GlideImageLoader
import com.zxjdev.smile.presentation.common.util.image.ImageLoader

import dagger.android.support.DaggerFragment

open class BaseFragment : DaggerFragment() {

    protected val imageLoader: ImageLoader by lazy {
        GlideImageLoader(this)
    }
}
